magento/magento2
# ISSUE-28814 - Ensure MSRP info on gesture always works

### Description (*)
When MSRP is enabled and a product is set to show price on gesture, the MSRP javascript relies on an MSRP popup template being present on the page (<script data-role="msrp-popup-template" type="text/x-magento-template">). If that template is not present, there is a js error and the info popup won't be shown on click.

The MSRP module adds the popup template to layout handles where products are expected (category view, product view, cart, etc.). However, products can be added to virtually any page via CMS content / widgets, so any page outside the expected list will not have gesture functionality.

This PR removes the popup from specific layout handles and instead adds it to default so it's rendered on every page. However, an ifconfig attribute is added to the popup block so that it will only be processed if MSRP is enabled.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#28814

### Manual testing scenarios (*)

1. Enable MSRP in admin configuration (Sales > Minimum Advertised Price > Enable MAP).
2. Edit a product. In advanced pricing, set a MAP price greater than the product's price and set Display Actual Price to On Gesture.
3. Visit a category containing that product. Click "Click for price" and confirm it shows the price information. (this was working before and should continue to work)
4. In admin, edit a CMS page and insert a Catalog Products List widget. Set the conditions so the MAP product will be displayed in the list and save the CMS page.
5. Visit the CMS page on the store front. Click "Click for price" on the product and confirm it shows the price information. (this was NO working before)


### Questions or comments
There were several options to fix this issue:
1. Consider support for MSRP on additional pages (such as non-home CMS pages) to be beyond the scope of the native application. Merchants must add theme customizations to add the popup to layout handles where they desire to show products with MSRP. **I consider this viewpoint invalid. The feature is broken as long as "Click for price" is present but doesn't work.**
2. Attempt some kind of registry model to track whether an MSRP block has been rendered for a product. Add a new block at the bottom of the page that checks the registry and renders the popup block if MSRP was ever rendered. This will ensure that the popup block is output when MSRP is present on the page, and not output if there was no MSRP. **This is overly complicated and block caching could produce scenarios where MSRP is rendered in cached content but the popup template is not rendered.**
3. Load the template via js / ajax. This could be done always (instead of the current behavior of embedding it in the page), or as a fallback if the popup isn't already found on the page (such as non-home CMS pages).

In the end, none of these seemed like good solutions compared to just adding the block to every page.

Additional note: Issue #28814 mentions a second case where MAP gesture doesn't work; any time the same MAP product is displayed twice on the same page. I was unable to reproduce that use case.

### Contribution checklist (*)
